### PR TITLE
Build Styx Docker image using fabric8 docker maven plugin 

### PR DESCRIPTION
**Pull request by mikkokar**
This PR adds `fabric8` docker maven plugin to build Styx Core docker image.

This is to simplify and "abstract away" the details how to build Styx Docker image automatically from other projects.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -133,8 +133,4 @@ changelog:
 # Default configuration file: /styx/default-config/default.yml
 #
 docker-image: clean
-	mvn install -Prelease,linux -Dmaven.test.skip=true
-	mkdir -p ${DOCKER_CONTEXT}
-	cp `find  distribution/target -maxdepth 1 -name "styx*linux-x86_64.zip"` ${DOCKER_CONTEXT}/styx.zip
-	cp docker/styx-image/* ${DOCKER_CONTEXT}
-	docker build -t styxcore:latest --build-arg STYX_IMAGE=styx.zip ${DOCKER_CONTEXT}/.
+	mvn clean install -Prelease,linux,docker -Dmaven.test.skip=true
```
Comment (dvlato) [inline Makefile]: do we need mvn clean even though there's a 'clean' above?

File: distribution/pom.xml [hunk 2]
```diff
@@ -133,4 +137,58 @@
...
+              </execution>
+            </executions>
+
+            <configuration>
+              <images>
+                <image>
+                  <alias>styxcore</alias>
+                  <name>${styxcore.docker.image}</name>
+                  <build>
+                    <dockerFile>${project.basedir}/../docker/styx-image/Dockerfile</dockerFile>
```
Comment (dvlato) [inline distribution/pom.xml]: we probably can reference the parent project here instead of using  ${project.basedir}/../

File: Makefile (hunk 1 above)
Comment (mikkokar) [reply Makefile]: No. Thanks for spotting 👍

File: distribution/pom.xml (hunk 2 above)
Comment (mikkokar) [reply distribution/pom.xml]: I have changed this to `${project.parent.basedir}`.

**Pull request merged**